### PR TITLE
More tweaks to regex source generator code

### PR DESCRIPTION
**Pull request by stephentoub**
These are minor tweaks to make the code more readable in various situations.
cc: @eerhardt

### CODE REVIEW

**Comment by drieseng** [comment]
@stephentoub Below are a few small questions that are not specific to this PR at, so let me know if you want me to a separate "issue" for these.

* Do source generators have to be public?
* If yes, will the **RegexGenerator** class become part of the (BCL) API surface? Won't this make it more cumbersome to introduce (breaking) changes (eg. if ever you'd want to implement another interface instead of **IIncrementalGenerator**).
* Will it ship as a separate (NuGet) package?

PS. I don't want to waste too much of your time. I'm happy with binary answers.

File: src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs [hunk 1]
```diff
@@ -263,13 +263,22 @@ private static void EmitFindFirstChar(IndentedTextWriter writer, RegexMethod rm,
...
-                $"if (runtextpos{minRequiredLengthOffset} >= runtextbeg)"))
+            Debug.Assert(rm.Code.Tree.MinRequiredLength >= 0);
+            string clause = !rtl ?
+                rm.Code.Tree.MinRequiredLength switch
+                {
+                    0 => "if (runtextpos <= runtextend)",
+                    1 => "if (runtextpos < runtextend)",
+                    _ => $"if (runtextpos < runtextend - {minRequiredLength - 1})"
+                } :
+                rm.Code.Tree.MinRequiredLength switch
```
Comment (eerhardt) [inline src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs:275]: Can we use the `minRequiredLength` variable?

```suggestion
            Debug.Assert(minRequiredLength  >= 0);
            string clause = !rtl ?
                minRequiredLength switch
                {
                    0 => "if (runtextpos <= runtextend)",
                    1 => "if (runtextpos < runtextend)",
                    _ => $"if (runtextpos < runtextend - {minRequiredLength - 1})"
                } :
                minRequiredLength switch
```

File: src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs [hunk 2]
```diff
@@ -1566,42 +1565,31 @@ void EmitSingleCharAtomicLoop(RegexNode node, bool emitLengthChecksIfRequired =
...
                             break;
                         case RegexNode.Notoneloopatomic:
                             expr = ToLowerIfNeeded(hasTextInfo, options, expr, IsCaseInsensitive(node) && RegexCharClass.ParticipatesInCaseConversion(node.Ch));
-                            expr = $"{expr} == {Literal(node.Ch)}";
+                            expr = $"{expr} != {Literal(node.Ch)}";
                             break;
                         case RegexNode.Setloopatomic:
                             expr = MatchCharacterClass(hasTextInfo, options, expr, node.Str!, IsCaseInsensitive(node));
-                            expr = $"!{expr}";
+                            expr = $"{expr}";
```
Comment (eerhardt) [inline src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs:1576]: Is this line doing anyting?

**Review by eerhardt — APPROVED** [review]
LGTM

File: src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs (hunk 2 above)
Comment (stephentoub) [reply src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs:1576]: Nope, it's a holdover from when it was (with the `!`), and then when I removed the inversion, I neglected to notice it was now a nop.  Good catch. I'll delete it.

**Comment by stephentoub** [comment]
> will the RegexGenerator class become part of the (BCL) API surface?

No, it'll be part of the SDK, just like analyzers.

> Will it ship as a separate (NuGet) package?

There are no plans to ship it in a separate package from the SDK.

**Pull request merged**